keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -115,45 +116,60 @@ class ApplicationService(
     val environmentSummaries = getEnvironmentSummariesFor(application)
 
     return deliveryConfig.artifacts.map { artifact ->
-      repository.artifactVersions(artifact).map { version ->
+      val artifactVersionSummaries = repository.artifactVersions(artifact).map { version ->
         val artifactSummariesInEnvironments = mutableSetOf<ArtifactSummaryInEnvironment>()
 
         environmentSummaries.forEach { environmentSummary ->
           val environment = deliveryConfig.environments.find { it.name == environmentSummary.name }!!
           environmentSummary.getArtifactPromotionStatus(artifact, version)?.let { status ->
-            if (status == PromotionStatus.PENDING) {
-              ArtifactSummaryInEnvironment(
+            var artEnvSummary = when (status) {
+              PENDING -> ArtifactSummaryInEnvironment(
                 environment = environmentSummary.name,
                 version = version,
                 state = status.name.toLowerCase()
               )
-            } else {
-              repository.getArtifactSummaryInEnvironment(
```

Comment:
This was the problem - we weren't correctly handling the skipped status, because when we went to get the summary from the database it said it had a `PENDING` status.